### PR TITLE
Fix issue in nerdctl rm.

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan <smahajan@roblox.com>

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
Issue:

Currently (on master HEAD), if I have a `running`, `paused` and `stopped` container. 
And I try to do `docker rm $(docker ps -aq)`

```
root@vagrant:~/go/src/github.com/nerdctl# docker ps -a
CONTAINER ID    IMAGE                             COMMAND        CREATED           STATUS                         PORTS    NAMES
03545c1d0c52    docker.io/library/ubuntu:18.04    "sleep 600"    26 seconds ago    Paused                                  c2
6d459455950b    docker.io/library/ubuntu:18.04    "sleep 600"    21 seconds ago    Stopped (137) 3 seconds ago             c3
86fbc4265da1    docker.io/library/ubuntu:18.04    "sleep 600"    30 seconds ago    Running                                 c1
root@vagrant:~/go/src/github.com/nerdctl# docker rm $(docker ps -aq)
WARN[0000] failed to remove container "03545c1d0c52b9daf2add34bd94b96f3352288243108bf4ebfcec76db1f7d2e6"  error="cannot remove a paused container 03545c1d0c52b9daf2add34bd94b96f3352288243108bf4ebfcec76db1f7d2e6"
WARN[0000] failed to remove container "03545c1d0c52b9daf2add34bd94b96f3352288243108bf4ebfcec76db1f7d2e6"  error="cannot remove a paused container 03545c1d0c52b9daf2add34bd94b96f3352288243108bf4ebfcec76db1f7d2e6"
FATA[0000] cannot remove a paused container 03545c1d0c52b9daf2add34bd94b96f3352288243108bf4ebfcec76db1f7d2e6
```

It will error out if it is not able to remove one of the containers e.g. in this case, it is not able to remove the `paused` container since no `force` flag was set.

The correct behavior should be to remove the containers it can (Stopped), and print errors on the containers it can't  (Paused, running).

This PR fixes this issue.

```
root@vagrant:~/go/src/github.com/nerdctl# docker ps -a
CONTAINER ID    IMAGE                             COMMAND        CREATED          STATUS                         PORTS    NAMES
03545c1d0c52    docker.io/library/ubuntu:18.04    "sleep 600"    4 minutes ago    Paused                                  c2
6d459455950b    docker.io/library/ubuntu:18.04    "sleep 600"    4 minutes ago    Stopped (137) 3 minutes ago             c3
86fbc4265da1    docker.io/library/ubuntu:18.04    "sleep 600"    4 minutes ago    Running                                 c1
root@vagrant:~/go/src/github.com/nerdctl# docker rm $(docker ps -aq)
Error response from daemon: You cannot remove a paused container 03545c1d0c52b9daf2add34bd94b96f3352288243108bf4ebfcec76db1f7d2e6. Unpause the container before attempting removal or force remove
6d459455950b
Error response from daemon: You cannot remove a running container 86fbc4265da1af30732ae8821e3d6e5e08e5cfb760485239db30957f7bd98223. Stop the container before attempting removal or force remove
```

File: rm.go [hunk 1]
```diff
@@ -144,9 +141,18 @@ func removeContainer(ctx context.Context, client *containerd.Client, id string,
...
-			return errors.Errorf("cannot remove a %v container %v", status.Status, id)
+			var msg string
+			if status.Status == containerd.Paused || status.Status == containerd.Running {
+				if status.Status == containerd.Paused {
+					msg = "Unpause"
+				} else if status.Status == containerd.Running {
+					msg = "Stop"
+				}
+
+				_, err := fmt.Fprintf(clicontext.App.Writer, "Error response from daemon: You cannot remove a %v container %v. %s the container before attempting removal or force remove\n", status.Status, id, msg)
```
Comment (AkihiroSuda) [inline rm.go:152]: Not really "Error response from daemon". I don't think the error message needs to 100% correspond to Docker.

File: rm.go [hunk 2]
```diff
@@ -144,9 +141,18 @@ func removeContainer(ctx context.Context, client *containerd.Client, id string,
 		}
 	default:
 		if !force {
-			return errors.Errorf("cannot remove a %v container %v", status.Status, id)
+			var msg string
+			if status.Status == containerd.Paused || status.Status == containerd.Running {
```
Comment (AkihiroSuda) [inline rm.go:145]: Can we move these comparison to `switch`?

File: rm.go (hunk 1 above)
Comment (shishir-a412ed) [reply rm.go:152]: I agree it doesn't need to be 100% matching to docker 🙂 
I felt it just makes it easier for someone who is transitioning from docker CLI to `nerdctl`

Either way, let me know what do you want me to change it to? and I will update the message.

Comment (AkihiroSuda) [reply rm.go:152]: I think we can just remove ""Error response from daemon: " part.

Comment (shishir-a412ed) [reply rm.go]: Fixed.

File: rm.go (hunk 2 above)
Comment (shishir-a412ed) [reply rm.go]: Fixed.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**